WordPress/gutenberg
# Use available font weights and styles in FontAppearanceControl

## What?

Currently in Global Styles > Typography, the font Appearance dropdown includes a static list of options that is hard-coded in the component. Often many of these options aren't available to actually use, depending on the installed font families and the active font. 

This PR swaps out this list of static options and replaces them with a list of available font weights/styles based on the currently active font.

## Why?

Fixes https://github.com/WordPress/gutenberg/issues/49090.

## How?

Passes the list of font faces from the active font family to the `FontAppearanceControl` component. This can then be used instead of the static `FONT_WEIGHTS` and `FONT_STYLES` constants currently used to populate the Appearance dropdown menu.

## Testing Instructions




1. Open Global Styles > Typography
2. Open the Appearance dropdown menu
3. The menu should include only the weights and styles made available by the currently active font
4. Additionally, Bold is added as an Appearance option if the font doesn't already have a weight option of 600 or above, along with an italic option for each available weight
5. Change the active font by using the Font dropdown menu (ideally to a font that has different weights/styles)
6. Ensure that the Appearance dropdown menu is updated to reflect the new font

Please also test changing to a system font and a variable font (e.g. AR One Sans). System fonts should have all possible Appearance options, and variable fonts should have the range of weights included with the font shown in the Appearance dropdown.

## Screenshots or screencast 

| Before | After |
| ----- | ----- |
| <img width="466" alt="Screenshot 2024-05-24 at 00 25 38" src="https://github.com/WordPress/gutenberg/assets/1645628/e1e416b9-39a6-4a93-8830-86be22ceab67"> | <img width="466" alt="Screenshot 2024-05-24 at 00 15 37" src="https://github.com/WordPress/gutenberg/assets/1645628/7061b54e-6caf-4237-9489-205e2152fe31"> |
